File: turi/src/main/java/com/turi/touristicplace/domain/model/TouristicPlace.java
package com.turi.touristicplace.domain.model;

import com.turi.touristicplace.infrastructure.adapter.repository.TouristicPlaceEntity;
import lombok.*;

import java.time.LocalTime;

@Getter
@Setter
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
@Builder(setterPrefix = "with")
public final class TouristicPlace
{
    private Long touristicPlaceId;
    private Long premiumId;
    private Long addressId;
    private TouristicPlaceType touristicPlaceType;
    private String name;
    private String description;
    private String information;
    private String ownerDescription;
    private LocalTime checkInTimeFrom;
    private LocalTime checkInTimeTo;
    private LocalTime checkOutTimeFrom;
    private LocalTime checkOutTimeTo;
    private boolean prepayment;
    private int cancelReservationDays;

    public static TouristicPlace of(final TouristicPlaceEntity entity)
    {
        return TouristicPlace.builder()
                .withTouristicPlaceId(entity.getTouristicPlaceId())
                .withPremiumId(entity.getPremiumId())
                .withAddressId(entity.getAddressId())
                .withTouristicPlaceType(TouristicPlaceType.fromValue(entity.getTouristicPlaceType()))
                .withName(entity.getName())
                .withDescription(entity.getDescription())
                .withInformation(entity.getInformation())
                .withOwnerDescription(entity.getOwnerDescription())
                .withCheckInTimeFrom(entity.getCheckInTimeFrom())
                .withCheckInTimeTo(entity.getCheckInTimeTo())
                .withCheckOutTimeFrom(entity.getCheckOutTimeFrom())
                .withCheckOutTimeTo(entity.getCheckOutTimeTo())
                .withPrepayment(entity.isPrepayment())
                .withCancelReservationDays(entity.getCancelReservationDays())
                .build();
    }
}

File: turi-ui/src/types/touristicPlace.ts
export interface TouristicPlace {
    readonly touristicPlaceId?: number
    premiumId: number
    addressId?: number
    touristicPlaceType?: TouristicPlaceType
    name?: string
    description?: string
    information?: string
    ownerDescription?: string
    checkInTimeFrom?: string
    checkInTimeTo?: string
    checkOutTimeFrom?: string
    checkOutTimeTo?: string
    prepayment?: boolean
    cancelReservationDays?: number
}

export interface GuaranteedService {
    readonly guaranteedServiceId?: number
    touristicPlaceId: number
    service: string
}

export enum TouristicPlaceType {
    UNASSIGNED = 'UNASSIGNED',
    GUESTHOUSE = 'GUESTHOUSE',
    APARTMENT = 'APARTMENT',
    COTTAGES = 'COTTAGES',
    HOTEL = 'HOTEL',
    BB = 'BB',
    HOSTEL = 'HOSTEL'
}
